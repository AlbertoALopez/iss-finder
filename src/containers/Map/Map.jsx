import React, { Component } from 'react';
import { Segment } from 'semantic-ui-react';
import api from '../../utils/api';

export default class Map extends Component {
  constructor() {
    super();
    this.state = {
      issLat: '',
      issLon: '',
    };
  }

  componentDidMount() {
    const that = this;
    let issLat;
    let issLon;
    api.getISSPosition()
      .then((response) => {
        return response.json();
      })
      .then((json) => {
        issLat = json.iss_position.latitude;
        issLon = json.iss_position.longitude;
        that.setState({
          issLat,
          issLon,
        });
      })
      .catch((error) => {
        console.log(`Error fetching ISS data: ${error}`);
      });
  }

  render() {
    return (
      <Segment>
        {this.state.issLat}
        {this.state.issLon}
      </Segment>
    );
  }
}
