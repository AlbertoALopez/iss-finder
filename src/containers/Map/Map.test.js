import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Map from './Map.jsx';

test('Map container renders correctly', () => {
  const component = shallow(
    <Map />
  );

  expect(toJson(component)).toMatchSnapshot();
});