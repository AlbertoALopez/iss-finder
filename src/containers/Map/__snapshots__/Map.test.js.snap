exports[`test Map container renders correctly 1`] = `
<Segment>
  <LeafletMap
    issLat=""
    issLon="" />
</Segment>
`;
